### PR TITLE
DM-40295: Force mkl to retain binary compatibility.

### PR DESCRIPTION
**Pull request by ktlim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -145,6 +145,7 @@ outputs:
         - lmfit
         - lsstdesc.coord
         - matplotlib-base !=3.6.0
+        - mkl
```
Comment (erykoff) [inline recipe/meta.yaml:148]: Needs to be linux 64 only

Comment (ktlim) [reply recipe/meta.yaml:148]: OK, thanks.

**Comment by ktlim** [comment]
@conda-forge-admin, please rerender

**Review by erykoff — APPROVED** [review]
I believe this is correct.  But I am also wondering why we didn't get mkl with osx-64, but something something PyTorch I imagine.

**Comment by beckermr** [comment]
Osx PyTorch is only openblas iirc

**Pull request merged**